class-gf-razorpay.php

```diff
@@ -0,0 +1,336 @@
+    protected $_path                     = 'razorpay-gravity-forms/razorpay.php';
+    protected $_full_path                = __FILE__;
+    protected $_url                      = 'http://www.gravityforms.com';
+    protected $_title                    = 'Gravity Forms Razorpay Add-On';
+    protected $_short_title              = 'Razorpay';
+    protected $_supports_callbacks       = true;
+
+    // Permissions
+    protected $_capabilities_settings_page = 'gravityforms_razorpay';
+    protected $_capabilities_form_settings = 'gravityforms_razorpay';
+    protected $_capabilities_uninstall     = 'gravityforms_razorpay_uninstall';
+
+    // Automatic upgrade enabled
+    protected $_enable_rg_autoupgrade = true;
+
+    private static $_instance = null;
+
+    const GF_RAZORPAY_KEY    = 'gf_razorpay_key';
+    const GF_RAZORPAY_SECRET = 'gf_razorpay_secret';
+    const RAZORPAY_ENTRY_ID  = 'gf_razorpay_entry_id';
+    const RAZORPAY_ORDER_ID  = 'razorpay_order_id';
+
+    //cookie set for one day
+    const COOKIE_DURATION    = 86400;
+
+    public static function get_instance()
+    {
+        if (self::$_instance === null)
+        {
+            self::$_instance = new GFRazorpay();
+        }
+
+        return self::$_instance;
+    }
+
+    public function init_frontend()
+    {
+        parent::init_frontend();
+
+        add_action('gform_after_submission', array($this, 'pay_using_razorpay'), 10, 2);
+    }
+
+    public function plugin_settings_fields()
+    {
+        return array(
+            array(
+                'title'       => 'razorpay_settings',
+                'fields'      => array(
+                    array(
+                        'name'    => self::GF_RAZORPAY_KEY,
+                        'label'   => esc_html__('Razorpay Key', 'razorpay-gravity-forms'),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'name'    => self::GF_RAZORPAY_SECRET,
+                        'label'   => esc_html__('Razorpay Secret', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__('Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
+            ),
+        );
+    }
+
+    public function get_customer_fields_array($feed, $entry)
+    {
+        $fields = array();
+
+        $customerFields = $this->get_customer_fields();
+
+        foreach ($customerFields as $field)
+        {
+            $fieldId = $feed['meta'][$field['meta_name']];
+
+            $value = rgar($entry, $fieldId);
+
+            $fields[$field['meta_name']] = $value;
+        }
+
+        return $fields;
```

Comment:
@riagarg5 http://stackoverflow.com/a/694847/3160368

We use this regularly in our api code. Kindly refer to any of our netbanking gateways, For eg. `getVerifyAttributes` method in Federal. Please change.